src/rate_limiter/src/lib.rs

```diff
@@ -419,6 +419,7 @@ impl RateLimiter {
                     // order to enforce the bandwidth limit we need to prevent
                     // further calls to the rate limiter for
                     // `ratio * refill_time` milliseconds.
+                    #[allow(clippy::cast_sign_loss)] // ratio is always positive
```

Comment:
`u64::try_from(..).unwrap()` can be used here.

Reply:
I don't think u64::tryfrom(f64) exists

Reply:
You are right.

Reply:
What is the plan here?

Reply:
It seems for now the allow is necessary. So your current implementation is good.